docs/website/blog/2023-06-27-odo-v3.12.0.md

```diff
@@ -0,0 +1,211 @@
+---
+title: odo v3.12.0 Released
+author: Armel Soro
+author_url: https://github.com/rm3l
+author_image_url: https://github.com/rm3l.png
+image: https://raw.githubusercontent.com/redhat-developer/odo/main/docs/website/static/img/logo.png
+tags: ["release"]
+slug: odo-v3.12.0
+---
+
+odo `v3.12.0` is now out!
+
+<!--truncate-->
+
+To install `odo`, follow [the installation guide](../docs/overview/installation).
+
+## Notable Changes
+Check these playlists for an overview of the most notable changes in this release:
+- [Sprint 239](https://youtube.com/playlist?list=PLGMB2PY4SNOp4BahHQMM0itX95TL-WXPP)
+- [`odo v3.12.0` (Sprint 240)](https://youtube.com/playlist?list=PLGMB2PY4SNOpRVyAmNcqgaZ08DuurOyFq)
+
+### Features
+
+#### Nightly binaries
+
+To make it easier to try out new features of `odo`, you no longer need to wait until a new version is released.
+We now publish nightly builds of `odo` with any changes that have been merged into the `main` branch at that time.
+
+:::caution
+Bear in mind that such builds are provided as is and might be highly unstable.
+:::
+
+Find out more on the [installation page](/docs/overview/installation#nightly-builds) for instructions on how to download and install nightly builds.
+
+#### Full control over the application lifecycle with `odo dev --no-commands` and `odo run` 
+
+Prior to v3.12.0, `odo dev` would implicitly run the `build` and `run`/`debug` commands automatically, and users had no control over the commands execution.
+To give users more control over the lifecycle of their applications, we wanted to make it possible to not only start a Dev Session without any assumptions on the commands,
+but also to run any command manually at any time.
+
+The new `--no-commands` option allows to start a Dev Session without implicitly running the `build` and `run`/`debug` commands,
+and the new `odo run` command allows to run any command from the Devfile against an already-running Dev session.
+
+To make the most out of this workflow, users can:
+1. start a Dev session without executing any commands, with [`odo dev --commands`](/docs/command-reference/dev#running-with-no-commands)
```

Comment:
```suggestion
1. start a Dev session without executing any commands, with [`odo dev --no-commands`](/docs/command-reference/dev#running-with-no-commands)
```

Reply:
Whoops, good catch - thanks. My keyboard seems to forget to type some words :)